File: C#/575. Distribute Candies.cs
public class Solution {
    public int DistributeCandies(int[] candyType) {
        int n = candyType.Length;
        Dictionary<int, int> dict = new Dictionary<int, int>();
        int res = 0;
        foreach(int i in candyType) {
            if (!dict.ContainsKey(i)) {
                if (res == n/2) {
                    return n/2;
                }
                res++;
            }
            dict[i]=1;
        }
        return res;
    }
}

File: C++/575. Distribute Candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        int n = candyType.size();
        unordered_map<int, int> mp;
        int res = 0;
        for (auto i: candyType) {
            if (mp[i]==0) {
                if (res == n/2) {
                    return n/2;
                }
                res++;
            }
            mp[i]++;
        }
        return res;
    }
};
